src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -689,6 +727,68 @@ private static bool StringRequiresHtmlDecoding(string s)
 
 #endregion
 
+        private struct PooledStringBuilder
```

Comment:
We have a helper class like this here: https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Text/ValueStringBuilder.cs that has a few more optimizations. Try using that instead.

Reply:
It looks like `ValueStringBuilder` is backed by a `Span` - what's the best way of writing a `Span<char>` to a `TextWriter`? I don't want to just loop over the span and call `Write` because that'll be slow.

Reply:
`void Write(System.ReadOnlySpan<char> buffer)` method.